websites/R/Roblox/presence.ts

```diff
@@ -460,201 +440,243 @@ presence.on("UpdateData", async () => {
+					!pathname.includes("/settings"): {
+					const Id = pathname.split("/")[2];
+					if (talentUserData[0] !== Id) {
+						talentUserData = [
+							Id,
+							document.head.title.replace("'s Creator Page - Talent Hub", ""),
+						];
+						const req = await fetch(
+							`https://users.roblox.com/v1/users/${Id}`
+						).then(response => response.json());
+						talentUserData = [Id, req.name ?? talentUserData[1]];
+					}
+					if (!talentUserData[1]) presenceData.state = "Browsing Profile";
+					else presenceData.state = `Looking at ${talentUserData[1]}'s Profile`;
+					presenceData.largeImageKey = Id
+						? `https://www.roblox.com/Thumbs/Avatar.ashx?x=420&y=420&userid=${Id}`
+						: Assets.DeveloperLogo;
+
 					presenceData.buttons = [
 						{
 							label: "Visit Profile",
-							url: document.URL,
+							url: href,
 						},
 					];
+					break;
 				}
-			} else if (
-				pathname.includes("/jobs/") &&
-				document.querySelector('[data-testid="job-view-header"]')
-			) {
-				presenceData.state = `Looking at Job Post: ${
-					document.querySelector<HTMLParagraphElement>(
-						'[data-testid="job-view-header"] .MuiTypography-root.MuiTypography-body1'
-					).textContent
-				}`;
-
-				if (buttons) {
+				case pathname.includes("/jobs/") &&
+					!!document.querySelector('[data-testid="job-view-header"]'): {
+					presenceData.state = `Looking at Job Post: ${
+						document.querySelector<HTMLParagraphElement>(
+							'[data-testid="job-view-header"] .MuiTypography-root.MuiTypography-body1'
+						).textContent
+					}`;
+
 					presenceData.buttons = [
 						{
 							label: "Visit Job",
-							url: document.URL,
+							url: href,
 						},
 					];
+					break;
+				}
+				case pathname.includes("/inbox"): {
+					presenceData.state = "Browsing Inbox";
+					break;
 				}
-			} else if (pathname.includes("/inbox"))
-				presenceData.state = "Browsing Inbox";
-			else if (pathname.includes("/settings"))
-				presenceData.state = "Editing Settings";
-			else if (pathname.includes("/jobs/create"))
-				presenceData.state = "Creating Job Post";
+
+				case pathname.includes("/settings"): {
+					presenceData.state = "Editing Settings";
+					break;
+				}
+				case pathname.includes("/jobs/create"): {
+					presenceData.state = "Creating Job Post";
+					break;
+				}
+			}
+
 			break;
 		}
 	}
 
+	if (!buttons && presenceData.buttons) delete presenceData.buttons;
+	if (
+		!imagesEnabled &&
+		presenceData.largeImageKey !== Assets.Logo &&
+		!hostname.includes("devforum") // ImagesEnabled setting off & The largeimagekey isnt Assets.Logo & Its not the devforum
+	)
+		presenceData.largeImageKey = Assets.Logo;
+	else if (
+		!imagesEnabled &&
+		presenceData.largeImageKey !== Assets.DeveloperLogo &&
+		hostname.includes("devforum") // ImagesEnabled setting off & The largeimagekey isnt Assets.DeveloperLogo & Its the devforum
+	)
+		presenceData.largeImageKey = Assets.DeveloperLogo;
+
```

Comment:
Need to change this since developerlogo is also used outside the developerforum thingy